src/__experimental__/components/date/date-picker.component.js

```diff
@@ -44,10 +45,11 @@ const DatePicker = (props) => {
   };
 
   useEffect(() => {
-    if (props.selectedDate && monthOrYearHasChanged(datepicker, props.selectedDate)) {
-      datepicker.current.showMonth(props.selectedDate);
+    if (hasComponentUpdated(currentInputDate, isoFormattedValueString(props.inputDate))) {
+      datepicker.current.showMonth(DateHelper.stringToDate(isoFormattedValueString(props.inputDate)));
```

Comment:
`DateHelper.stringToDate(isoFormattedValueString(props.inputDate));` could be a const to increase readability

Reply:
Agreed 👍